pkg/reconciler/namespace/namespace.go

```diff
@@ -237,3 +256,43 @@ func (r *Reconciler) reconcileBroker(ctx context.Context, ns *corev1.Namespace)
 	// Don't update anything that is already present.
 	return current, nil
 }
+
+func CopySecret(r *Reconciler, srcNS string, srcSecretName string, tgtNS string, svcAccount string) (*corev1.Secret, error) {
+	tgtNamespaceSvcAcct := r.KubeClientSet.CoreV1().ServiceAccounts(tgtNS)
+	srcSecrets := r.KubeClientSet.CoreV1().Secrets(srcNS)
+	tgtNamespaceSecrets := r.KubeClientSet.CoreV1().Secrets(tgtNS)
+
+	// First try to find the secret we're supposed to copy
+	srcSecret, err := srcSecrets.Get(srcSecretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	// check for nil source secret
+	if srcSecret == nil {
+		return nil, errors.New("error copying secret")
```

Comment:
Could we log a little better error here. <...time passes, I remember seeing this before...>
Perhaps we should move the function from test/common/test_runner.go to pkg instead and use it here and in testing so that we don't need two identical copies?
https://github.com/knative/eventing/blob/master/test/common/test_runner.go#L137

Maybe pkg/utils/secret.go

Reply:
Agree - just moved it to pkg/utils/secret.go & added some test cases.